src/main/scala/com.gu.contentapi.client/GuardianContentClient.scala

```diff
@@ -3,29 +3,21 @@ package com.gu.contentapi.client
 import com.gu.contentapi.client.model._
 import com.gu.contentapi.client.parser.JsonParser
 import com.gu.contentapi.client.utils.QueryStringParams
-import dispatch.{FunctionHandler, Http}
 import com.gu.contentapi.buildinfo.BuildInfo
 
 import scala.concurrent.{ExecutionContext, Future}
 
 case class GuardianContentApiError(httpStatus: Int, httpMessage: String) extends Exception(httpMessage)
 
-trait ContentApiClientLogic {
-  val apiKey: String
+case class HttpResponse(body: String, statusCode: Int, statusMessage: String)
 
-  private val userAgent = "content-api-scala-client/"+BuildInfo.version
+trait HttpClient {
+  def get(url: String, headers: Map[String, String])(implicit context: ExecutionContext): Future[HttpResponse]
```

Comment:
Headers should be List[(String, String)] - ordered and multiple values for each header name